Tests/BSONTests/BSONCorpusTests.swift

```diff
@@ -129,97 +163,93 @@ final class BSONCorpusTests: BSONTestCase {
                     let docFromNative = BSONDocument(fromArray: nativeFromDoc)
                     expect(docFromNative.toByteString()).to(equal(cBData.toByteString()))
 
-                    if testFile.description == "Decimal128" {
-                        // TODO: This should be tested by EXTJSON
-                        struct Decimal128CanonicalExtJSON: Codable {
-                            struct Value: Codable {
-                                enum CodingKeys: String, CodingKey {
-                                    case numberDecimal = "$numberDecimal"
-                                }
-
-                                var numberDecimal: String
-                            }
+                    // native_to_canonical_extended_json( bson_to_native(cB) ) = cEJ
+                    let canonicalEncoder = ExtendedJSONEncoder()
+                    canonicalEncoder.mode = .canonical
+                    expect(try canonicalEncoder.encode(docFromCB))
+                        .to(cleanEqual(test.canonicalExtJSON), description: test.description)
 
-                            var d: Value
-                        }
-                        let extjson = test.canonicalExtJSON.data(using: .ascii)!
-                        let jsonResult = try JSONDecoder().decode(Decimal128CanonicalExtJSON.self, from: extjson)
-                        let decimal128CorpusString = jsonResult.d.numberDecimal
+                    // native_to_relaxed_extended_json( bson_to_native(cB) ) = rEJ (if rEJ exists)
+                    let relaxedEncoder = ExtendedJSONEncoder() // default mode is .relaxed
+                    if let rEJ = test.relaxedExtJSON {
+                        expect(try relaxedEncoder.encode(docFromCB))
+                            .to(cleanEqual(rEJ), description: test.description)
+                    }
 
-                        let decimal128FromString = try BSONDecimal128(decimal128CorpusString)
-                        let decimal128FromBinary = docFromCB.d!.decimal128Value!
+                    // for cEJ input:
+                    // native_to_canonical_extended_json( json_to_native(cEJ) ) = cEJ
+                    expect(try canonicalEncoder.encode(try decoder.decode(BSON.self, from: cEJData)))
```

Comment:
why BSON instead of BSONDocument here? I would think the data is always a document and not other BSON types?

Reply:
Yeah BSONDocument makes sense here, and probably in all of the places where I currently have `BSON.self`?

Reply:
yeah I believe so, sgtm